packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts

```diff
@@ -208,3 +261,14 @@ export function clientCertificatesToTLSOptions(
   }
   return tlsOptions;
 }
+
+function rewriteToLocalhostIfNeeded(host: string): string {
+  return host === 'local.playwright' ? 'localhost' : host;
+}
+
+// For testing.
+if (require.main === module) {
```

Comment:
I'd prefer to have this code in a separate file under `tests` or `utils` instead.